### PR TITLE
docs(firebase_analytics): example app initialization and docs support status

### LINKED ISSUE #7543: The Crashlytics build ID is missing
Implementation:
```
apply plugin: 'com.google.firebase.crashlytics'
...
implementation 'com.google.firebase:firebase-crashlytics:17.3.0'
...
classpath 'com.google.firebase:firebase-crashlytics-gradle:2.7.1'
...
runZonedGuarded(() {
    runApp(
      LocalizedApp(
        delegate,
        MyApp(),
      ),
    );
  }, FirebaseCrashlytics.instance.recordError);
```

firebase_crashlytics: ^2.4.3 

doctor:
```
[✓] Flutter (Channel stable, 2.5.3, on macOS 11.6.1 20G224 darwin-x64, locale en-GB)
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
[✓] Xcode - develop for iOS and macOS
[✓] Chrome - develop for the web
[✓] Android Studio (version 2020.3)
[✓] VS Code (version 1.62.0)
[✓] Connected device (3 available)

• No issues found!
```

### LINKED ISSUE #7639: 🐛 [firebase_analytics] After migrated to Dart-only initialization, analytics is not working on Android.
## Bug report

https://firebase.flutter.dev/docs/manual-installation#android

I migrated my app to Dart-only initialization.
Then, firebase analytics is not working on Android. I got this error. 

> E/FA      (17925): Missing google_app_id. Firebase Analytics disabled. See https://goo.gl/NAOOOI

### Steps to reproduce

1. flutterfire configure
2. Remove the /android/app/google-services.json file.
3. Remove the google-services dependency from the /android/build.gradle file.
4. Remove the plugin execution via the /android/app/build.gradle file.

```
import 'package:firebase_core/firebase_core.dart';
import 'firebase_options.dart';
...
void main() async {
  WidgetsFlutterBinding.ensureInitialized();

  await Firebase.initializeApp(options: DefaultFirebaseOptions.currentPlatform);

  runApp(const App());
}
```

### Expected behavior

firebase_analytics runs fine on android.

---

## Additional context

---

### Flutter doctor

Run `flutter doctor` and paste the output below:

<details><summary>Click To Expand</summary>

```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 2.8.1, on macOS 12.1 21C52 darwin-x64, locale
    ko-KR)
[!] Android toolchain - develop for Android devices (Android SDK version 32.0.0)
    ! Some Android licenses not accepted.  To resolve this, run: flutter doctor
      --android-licenses
[✓] Xcode - develop for iOS and macOS (Xcode 13.2)
[✓] Android Studio (version 2020.3)
[✓] VS Code (version 1.63.2)
[!] Connected device
    ! No devices available

! Doctor found issues in 2 categories.
```

</details>

---

### Flutter dependencies

Run `flutter pub deps -- --style=compact` and paste the output below:

<details><summary>Click To Expand</summary>

```
Dart SDK 2.15.1
Flutter SDK 2.8.1
hunmin 1.0.0+13

dependencies:
- audioplayers 0.20.1 [uuid path_provider http flutter flutter_web_plugins]
- firebase_analytics 9.0.4 [firebase_analytics_platform_interface firebase_analytics_web firebase_core firebase_core_platform_interface flutter]
- firebase_core 1.10.6 [firebase_core_platform_interface firebase_core_web flutter meta]
- flutter 0.0.0 [characters collection meta typed_data vector_math sky_engine]
- package_info_plus 1.3.0 [flutter package_info_plus_platform_interface package_info_plus_linux package_info_plus_macos package_info_plus_windows package_info_plus_web]
- url_launcher 6.0.17 [flutter meta url_launcher_android url_launcher_ios url_launcher_linux url_launcher_macos url_launcher_platform_interface url_launcher_web url_launcher_windows]

dev dependencies:
- flutter_launcher_icons 0.9.2 [args image path yaml]
- flutter_lints 1.0.4 [lints]
- flutter_test 0.0.0 [flutter test_api path fake_async clock stack_trace vector_math async boolean_selector characters charcode collection matcher meta source_span stream_channel string_scanner term_glyph typed_data]

transitive dependencies:
- archive 3.1.6 [crypto path]
- args 2.3.0
- async 2.8.2 [collection meta]
- boolean_selector 2.1.0 [source_span string_scanner]
- characters 1.2.0
- charcode 1.3.1
- clock 1.1.0
- collection 1.15.0
- crypto 3.0.1 [collection typed_data]
- fake_async 1.2.0 [clock collection]
- ffi 1.1.2
- file 6.1.2 [meta path]
- firebase_analytics_platform_interface 3.0.3 [firebase_core flutter meta plugin_platform_interface]
- firebase_analytics_web 0.4.0+4 [firebase_analytics_platform_interface firebase_core firebase_core_web flutter flutter_web_plugins js]
- firebase_core_platform_interface 4.2.3 [collection flutter meta plugin_platform_interface]
- firebase_core_web 1.5.3 [firebase_core_platform_interface flutter flutter_web_plugins js meta]
- flutter_web_plugins 0.0.0 [flutter js characters collection meta typed_data vector_math]
- http 0.13.4 [async http_parser meta path]
- http_parser 4.0.0 [charcode collection source_span string_scanner typed_data]
- image 3.1.0 [archive meta xml]
- js 0.6.3
- lints 1.0.1
- matcher 0.12.11 [stack_trace]
- meta 1.7.0
- package_info_plus_linux 1.0.3 [package_info_plus_platform_interface flutter path]
- package_info_plus_macos 1.3.0 [flutter]
- package_info_plus_platform_interface 1.0.2 [flutter meta plugin_platform_interface]
- package_info_plus_web 1.0.4 [flutter flutter_web_plugins http meta package_info_plus_platform_interface]
- package_info_plus_windows 1.0.4 [package_info_plus_platform_interface ffi flutter win32]
- path 1.8.0
- path_provider 2.0.8 [flutter path_provider_android path_provider_ios path_provider_linux path_provider_macos path_provider_platform_interface path_provider_windows]
- path_provider_android 2.0.9 [flutter path_provider_platform_interface]
- path_provider_ios 2.0.7 [flutter path_provider_platform_interface]
- path_provider_linux 2.1.4 [ffi flutter meta path path_provider_platform_interface xdg_directories]
- path_provider_macos 2.0.4 [flutter meta path_provider_platform_interface]
- path_provider_platform_interface 2.0.1 [flutter meta platform plugin_platform_interface]
- path_provider_windows 2.0.4 [ffi flutter meta path path_provider_platform_interface win32]
- petitparser 4.4.0 [meta]
- platform 3.1.0
- plugin_platform_interface 2.0.2 [meta]
- process 4.2.4 [file path platform]
- sky_engine 0.0.99
- source_span 1.8.1 [collection path term_glyph]
- stack_trace 1.10.0 [path]
- stream_channel 2.1.0 [async]
- string_scanner 1.1.0 [charcode source_span]
- term_glyph 1.2.0
- test_api 0.4.3 [async boolean_selector collection meta source_span stack_trace stream_channel string_scanner term_glyph matcher]
- typed_data 1.3.0 [collection]
- url_launcher_android 6.0.13 [flutter meta url_launcher_platform_interface]
- url_launcher_ios 6.0.13 [flutter meta url_launcher_platform_interface]
- url_launcher_linux 2.0.2 [flutter]
- url_launcher_macos 2.0.2 [flutter]
- url_launcher_platform_interface 2.0.4 [flutter plugin_platform_interface]
- url_launcher_web 2.0.5 [flutter flutter_web_plugins meta url_launcher_platform_interface]
- url_launcher_windows 2.0.2 [flutter]
- uuid 3.0.5 [crypto]
- vector_math 2.1.1
- win32 2.3.1 [ffi]
- xdg_directories 0.2.0 [meta path process]
- xml 5.3.1 [collection meta petitparser]
- yaml 3.1.0 [collection source_span string_scanner]
```

</details>

---

### PR DESCRIPTION
**Pull request by pr-Mais**
## Description

Fixed the initialization part in the example app for firebase_analytics and addd the support status section to show which plugins/platforms currently doesn't support the Dart-only initialization.

## Related Issues

closes #7639
closes #7543 

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [ ] All existing and new tests are passing.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] The analyzer (`melos run analyze`) does not report any problems on my PR.
- [ ] I read and followed the [Flutter Style Guide].
- [ ] I signed the [CLA].
- [ ] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change.
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/FirebaseExtended/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: docs/auth/social.mdx [hunk 1]
```diff
@@ -37,9 +37,13 @@ Install the official [`google_sign_in`](https://pub.dev/packages/google_sign_in)
 
 ```yaml title="pubspec.yaml"
 dependencies:
-  google_sign_in: "{{ external.google_sign_in }}"
+  google_sign_in: '{{ external.google_sign_in }}'
 ```
 
+:::caution
+Dart-only initialization isn't yet support by `google_sign_in` plugin, in which case you still have to do the [manual installation](./manual-installation.mdx) if you're using Google as an auth provider.
```
Comment (russellwheatley) [inline docs/auth/social.mdx:44]:
```suggestion
Dart-only initialization isn't yet supported by `google_sign_in` plugin, in which case you still have to do the [manual installation](./manual-installation.mdx) if you're using Google as an auth provider.
```

File: docs/cli.mdx [hunk 2]
```diff
@@ -20,6 +20,21 @@ Once installed, the `flutterfire` command will be globally available.
 
 ## Usage
 
+:::note
+Some plugins or platforms doesn't support Dart-only initialization **yet**, for which you will use the platform
```
Comment (russellwheatley) [inline docs/cli.mdx:24]:
```suggestion
Some plugins or platforms do not support Dart-only initialization **yet**, for which you will use the platform
```

File: docs/cli.mdx [hunk 3]
```diff
@@ -20,6 +20,21 @@ Once installed, the `flutterfire` command will be globally available.
...
+config files through the [manual installation](./manual-installation.mdx), the following table shows the current
```
Comment (russellwheatley) [inline docs/cli.mdx:25]:
```suggestion
config files through the [manual installation](./manual-installation.mdx). The following table shows the current
```

**Review by russellwheatley — CHANGES_REQUESTED** [review]
A few small changes from me. I think you'll also need to add back in the google service file for example/android as well.

**Comment by Ehesp** [comment]
Is the PR title correct here? It says fix but only applies docs and examples changes so shouldn't change versions of anything?

**Comment by pr-Mais** [comment]
yes I agree, `docs` could be?

File: docs/auth/social.mdx [hunk 4]
```diff
@@ -37,9 +37,13 @@ Install the official [`google_sign_in`](https://pub.dev/packages/google_sign_in)
 
 ```yaml title="pubspec.yaml"
 dependencies:
-  google_sign_in: "{{ external.google_sign_in }}"
+  google_sign_in: '{{ external.google_sign_in }}'
 ```
 
+:::caution
+Dart-only initialization isn't yet supported by `google_sign_in` plugin, in which case you still have to do the [manual installation](./manual-installation.mdx) if you're using Google as an auth provider.
```
Comment (Salakar) [inline docs/auth/social.mdx:44]:
```suggestion
Dart-only Firebase initialization isn't yet supported by `google_sign_in` plugin, if you're using Google as an auth provider you'll still have to do the [manual installation](./manual-installation.mdx) steps to integrate the Firebase configuration files.
```

**Pull request merged**